simexpal/launch/queue.py

```diff
@@ -1,18 +1,33 @@
 
+import os
+import tempfile
+
 from . import common
 from .. import queuesock
+from .. import util
 
 class QueueLauncher(common.Launcher):
-	def submit(self, config, run):
+	def submit(self, cfg, run):
+		util.try_mkdir(os.path.join(cfg.basedir, 'aux'))
+		util.try_mkdir(os.path.join(cfg.basedir, 'aux/_queue'))
+
 		if not common.lock_run(run):
 			return
 		common.create_run_file(run)
 
+		specfd, specfile = tempfile.mkstemp(prefix='', suffix='-spec.yml',
+											dir=os.path.join(cfg.basedir, 'aux/_queue'))
+		manifest = common.compile_manifest(run).yml
+		specs = {'manifest': manifest}
+		with os.fdopen(specfd, 'w') as f:
+			util.write_yaml_file(f, specs)
+
 		print("Launching experiment '{}', instance '{}' on local machine".format(
 				run.experiment.name, run.instance.shortname))
 
 		queuesock.sendrecv({
 			'action': 'launch',
-			'manifest': common.compile_manifest(run).yml
+			'manifest': manifest,
+			'specfile_path': specfile
```

Comment:
I dislike that we submit both the manifest and the specfile. We should not store information in a redundant way. We should probably just pass the file name and read the file from the daemon process.

Reply:
Done